ldelf/ta_elf.c

```diff
@@ -247,10 +257,20 @@ static void e32_save_symtab(struct ta_elf *elf, size_t tab_idx)
 	size_t str_idx = shdr[tab_idx].sh_link;
 
 	elf->dynsymtab = (void *)(shdr[tab_idx].sh_addr + elf->load_addr);
-	assert(!(shdr[tab_idx].sh_size % sizeof(Elf32_Sym)));
+	if (!ALIGNMENT_IS_OK(elf->dynsymtab, Elf32_Sym))
+		err(TEE_ERROR_GENERIC, "Bad alignment of dynsymtab %p",
+		    elf->dynsymtab);
+	check_range(elf, "dynsymtab", elf->dynsymtab, shdr[tab_idx].sh_size);
+
+	if (shdr[tab_idx].sh_size % sizeof(Elf32_Sym))
+		err(TEE_ERROR_GENERIC,
+		    "Size of dynsymtab not an even multiple of Elf32_Sym");
```

Comment:
very minor: suggest following (fit in a single line):
```C
		err(TEE_ERROR_GENERIC, "Bad alignment of dynsymtab size");
```

Reply:
Comment at wrong line?

Reply:
No, it's about "alignment of dynsymtab **size**".

Reply:
OK, thanks for clarifying.
As I said at another place, I'd prefer not making the debug printout more cryptic than necessary.